src/fpm/metadata/models.py

```diff
@@ -285,6 +302,9 @@ class Manifest(BaseModel):
 
     dev_dependencies: Dict[str, DependencyUnion] = Field({}, alias="dev-dependencies")
     """Development dependency meta data"""
+    
+    preprocess: Dict[Literal["cpp"], Preprocess] = {}
```

Comment:
Maybe this has to be a `str` rather than a `Literal["cpp"]` since it also provides more flexibility going forward.

Reply:
I am working on a solution for us to be able to test the schema on VS Code I'll let you know if I get it to work. FYI it will be a pain nonetheless

Reply:
We don't really have proper testing here other than verifying that it works for all existing fpm projects collected up to a certain date.

A better testing might be together to actually produce manifests and check whether fpm and this project agree on their validity and also their content. For the content I'm however unsure what a neutral format is that could be used for a comparison between what fpm sees and what we are seeing here.

Reply:
I was taking more the completions VS Code returns using the schema. That requires creating a dummy extension and attaching the generated JSON schema to it for validation (this I have done). The complication is that tomlValidation can only be handed a URL so we have to upload the fpm-manifest schema that we want to test in a local live server or something (that I am still figuring out).

For the general testing you were referring I think what we have is decent. TBH I wouldn't try to validate against fpm itself. Validating against existing fpm.tomls (or ones made by us) is good enough I think.